File: src/main/java/com/hopper/model/availability/agoda/Hotel.java
package com.hopper.model.availability.agoda;

import javax.xml.bind.annotation.XmlAccessType;
import javax.xml.bind.annotation.XmlAccessorType;
import javax.xml.bind.annotation.XmlElement;
import javax.xml.bind.annotation.XmlRootElement;

@XmlRootElement(name = "Hotel")
@XmlAccessorType(XmlAccessType.FIELD)
public class Hotel {

    @XmlElement(name = "Id")
    private int id;

    @XmlElement(name = "CheapestRoom")
    private CheapestRoom cheapestRoom;

    @XmlElement(name = "PaxSettings")
    private PaxSettings paxSettings;

    @XmlElement(name = "Rooms")
    private Rooms rooms;

    public PaxSettings getPaxSettings() {
        return paxSettings;
    }

    public void setPaxSettings(PaxSettings paxSettings) {
        this.paxSettings = paxSettings;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        id = id;
    }

    public CheapestRoom getCheapestRoom() {
        return cheapestRoom;
    }

    public void setCheapestRoom(CheapestRoom cheapestRoom) {
        cheapestRoom = cheapestRoom;
    }

    public Rooms getRoom() {
        return rooms;
    }

    public void setRoom(Rooms room) {
        this.rooms = room;
    }

    @Override
    public String toString() {
        return "Hotel{" +
                "Id=" + id +
                ", CheapestRoom=" + cheapestRoom +
                ", paxSettings=" + paxSettings +
                ", room=" + rooms +
                '}';
    }
}

File: src/main/scala/com/hopper/model/agoda/availability/response/Hotel.scala
package com.hopper.model.agoda.availability.response

import javax.xml.bind.annotation.{XmlAccessType, XmlAccessorType, XmlElement, XmlElementWrapper, XmlRootElement}

@XmlRootElement(name = "Hotel")
@XmlAccessorType(XmlAccessType.FIELD)
class Hotel
{
    @XmlElement(name = "Id")
    var id: String = _

    @XmlElement(name = "CheapestRoom")
    var cheapestRoom: CheapestRoom = _

    @XmlElement(name = "PaxSettings")
    var paxSettings: PaxSettings = _

    @XmlElementWrapper(name = "Rooms")
    @XmlElement(name = "Room")
    var rooms: Array[Room] = _

    override def toString = s"Hotel(id=$id, cheapestRoom=$cheapestRoom, paxSettings=$paxSettings, rooms=$rooms)"
}